test/test-TabTracker.js

```diff
@@ -0,0 +1,113 @@
+"use strict";
+
+const {before, after} = require("sdk/test/utils");
+const tabs = require("sdk/tabs");
+const ss = require("sdk/simple-storage");
+const {setTimeout} = require("sdk/timers");
+const {TabTracker} = require("lib/TabTracker");
+
+const NEWTAB_URL = "resource://activity-streams/data/content/activity-streams.html#/";
+
+exports.test_TabTracker_init = function(assert) {
+  assert.deepEqual(ss.storage.tabData, {}, "tabData storage starts out empty");
+};
+
+exports.test_TabTracker_open_close_tab = function*(assert) {
+  let tabClosedPromise = new Promise(resolve => {
+    let onOpen = function(tab) {
+      setTimeout(function() {
+        tab.close(() => {
+          tabs.removeListener("pageshow", onOpen);
+          resolve();
+        });
+      }, 5000);
```

Comment:
I was actually waiting 5 seconds to test that `totalTime` is set to a non-zero value. But waiting for page load and immediately closing might still end up with 0 `totalTime`. How about 10ms instead of 5000?